engine/baml-lib/llm-client/src/clients/aws_bedrock.rs

```diff
@@ -205,12 +199,33 @@ impl UnresolvedAwsBedrock {
                     None
                 }
             }
-            None => match ctx.get_env_var("AWS_SESSION_TOKEN") {
-                Ok(session_token) if !session_token.is_empty() => Some(session_token),
-                _ => None,
-            },
+            None => None,
         };
 
+        let (access_key_id, secret_access_key, session_token) =
+            match (access_key_id, secret_access_key, session_token) {
+                (None, None, None) => {
+                    // If no credentials provided, get them all from env vars
+                    let access_key_id = match ctx.get_env_var("AWS_ACCESS_KEY_ID") {
+                        Ok(key) if !key.is_empty() => Some(key),
+                        _ => None,
+                    };
+                    let secret_access_key = match ctx.get_env_var("AWS_SECRET_ACCESS_KEY") {
+                        Ok(key) if !key.is_empty() => Some(key),
+                        _ => None,
+                    };
+                    let session_token = match ctx.get_env_var("AWS_SESSION_TOKEN") {
+                        Ok(token) if !token.is_empty() => Some(token),
+                        _ => None,
+                    };
+                    (access_key_id, secret_access_key, session_token)
+                }
```

Comment:
@seawatts i suspect this can be collapsed as 
```suggestion
                (None, None, None) => {
                    // If no credentials provided, get them all from env vars
                    (ctx.get_env_var("AWS_ACCESS_KEY_ID").ok(), ctx.get_env_var("AWS_SECRET_ACCESS_KEY").ok(), ctx.get_env_var("AWS_SESSION_TOKEN").ok())
                }
```

but i don't want to put this through another round of testing right now so i'm leaving it as-is

(i'm not sure if the empty string difference is particularly significant here, i think it just triggers the fallback-to-env-var logic twice via two different paths